src/azul/indexer/document.py

```diff
@@ -222,23 +227,45 @@ def from_index(self, value: T) -> N:
 
 class PassThrough(Generic[T], FieldType[T, T]):
 
+    @property
+    @abstractmethod
+    def es_type(self) -> str:
+        return ''
+
     def to_index(self, value: T) -> T:
         return value
 
     def from_index(self, value: T) -> T:
         return value
 
 
-pass_thru_str: PassThrough[str] = PassThrough()
-pass_thru_int: PassThrough[int] = PassThrough()
-pass_thru_bool: PassThrough[bool] = PassThrough()
-pass_thru_json: PassThrough[JSON] = PassThrough()
+class PassThroughInt(PassThrough):
+    es_type = 'long'
+
+
+class PassThroughString(PassThrough):
+    es_type = 'string'
+
+
+class PassThroughBool(PassThrough):
+    es_type = 'boolean'
+
+
+class PassThroughJSON(PassThrough):
+    es_type = 'string'
```

Comment:
That makes no sense to me.

Reply:
@hannes-ucsc out of all the data types that can be used https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping-types.html `string` and `keyword` are the only types at successfully query Elasticsearch. It seems like `object` would be more appropriate here but a TransportError is reported:

```
elasticsearch.exceptions.TransportError: TransportError(500, 'search_phase_execution_exception', "class org.elasticsearch.index.mapper.ObjectMapper cannot be cast to class org.elasticsearch.index.mapper.FieldMapper (org.elasticsearch.index.mapper.ObjectMapper and org.elasticsearch.index.mapper.FieldMapper are in unnamed module of loader 'app')")
```

I suspect that it's because the fields are not populated yet that we get this transport error.

Reply:
For passthroughJSON the es_type should be set to None, the testing for these facets should be skipped. add a TODO for removal when passthroughJSON field types are changed to `nested` in both document.py and the test.